### PR TITLE
friends tab 

### PR DESCRIPTION
**Pull request by isabelleleeson**
Friends
also, fix get quests for children (just pass in empty array for now, still need to get request)
image

### CODE REVIEW

File: android/app/src/main/java/com/chorestory/fragment/ChildFriendsFragment.java [hunk 1]
```diff
@@ -3,18 +3,134 @@
...
+
+        childRecyclerView = view.findViewById(R.id.child_recycler_view);
+        addFriendButton = view.findViewById(R.id.add_friend_button);
+        addFriendEditText = view.findViewById(R.id.add_friend_edit_text);
+        childRecyclerView.setHasFixedSize(true);
+
+        childLayoutManager = new LinearLayoutManager(getActivity());
+        childRecyclerView.setLayoutManager(childLayoutManager);
+
+
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/fragment/ChildFriendsFragment.java]: add `addFriendButton` into `views`

File: android/app/src/main/java/com/chorestory/fragment/ChildFriendsFragment.java [hunk 2]
```diff
@@ -3,18 +3,134 @@
...
+        addFriendButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String username = addFriendEditText.getText().toString();
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/fragment/ChildFriendsFragment.java]: add `disableViews()` at the start on onClick
and add `enableViews()`in the appropriate places after

File: android/app/src/main/java/com/chorestory/fragment/ChildFriendsFragment.java [hunk 3]
```diff
@@ -3,18 +3,134 @@
...
+                    @Override
+                    public void onFailure(Call<ClanChildrenResponse> call, Throwable t) {
+                        // TODO: delete the token we have stored and redirect the user to the login page?
+                    }
+                });
+            } else {
+                // TODO: redirect to login page
+            }
+        }
+
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/fragment/ChildFriendsFragment.java]: delete line

File: android/app/src/main/java/com/chorestory/fragment/ChildQuestsFragment.java [hunk 4]
```diff
@@ -36,9 +39,21 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
     }
 
     private void setupViewPager() {
-        List<Fragment> fragmentList = Arrays.asList(new QuestsUpcomingFragment(),
-                new QuestsPendingFragment(),
-                new QuestsCompletedFragment());
+        Bundle bundle = new Bundle();
+        // TODO: make request to get quests
+        ArrayList<QuestParcelable> questParcelables = new ArrayList<>();
+        bundle.putParcelableArrayList("all_quests", questParcelables);
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/fragment/ChildQuestsFragment.java]: make "all_quests" into a string resource

File: android/app/src/main/res/layout/fragment_child_friends.xml [hunk 5]
```diff
@@ -6,4 +6,34 @@
     android:layout_height="match_parent"
     tools:context=".fragment.ChildFriendsFragment">
 
+    <EditText
```
Comment (jimzhaozhengwang) [inline android/app/src/main/res/layout/fragment_child_friends.xml]: `Reformat Code` on all files

File: android/app/src/main/java/com/chorestory/fragment/ChildFriendsFragment.java [hunk 6]
```diff
@@ -1,20 +1,144 @@
...
+                        if (response.isSuccessful() && response.body() != null && response.body().hasResponse()) {
+                            childrenList = response.body().getChildren();
+                            childAdapter = new ChildRecyclerViewAdapter(childrenList);
+                            childRecyclerView.setAdapter(childAdapter);
+                        }
+                    }
+
+                    @Override
+                    public void onFailure(Call<ClanChildrenResponse> call, Throwable t) {
+                        // TODO: delete the token we have stored and redirect the user to the login page?
```
Comment (keller00) [inline android/app/src/main/java/com/chorestory/fragment/ChildFriendsFragment.java]: What if request times out? Do we then sign people out for no reason?

File: android/app/src/main/java/com/chorestory/fragment/ChildFriendsFragment.java [hunk 7]
```diff
@@ -1,20 +1,144 @@
...
+                    }
+                });
+            } else {
+                // TODO: redirect to login page
```
Comment (keller00) [inline android/app/src/main/java/com/chorestory/fragment/ChildFriendsFragment.java]: I feel like I have seen this TODO so many times, we should have this as a helper and import it around, despite what @jimzhaozhengwang says making a function public and importing it is okay.

File: android/app/src/main/java/com/chorestory/fragment/ChildFriendsFragment.java (hunk 6 above)
Comment (isabelleleeson) [reply android/app/src/main/java/com/chorestory/fragment/ChildFriendsFragment.java]: good point ill just add a toast with Internal error occurred.

File: android/app/src/main/java/com/chorestory/fragment/ChildFriendsFragment.java (hunk 7 above)
Comment (isabelleleeson) [reply android/app/src/main/java/com/chorestory/fragment/ChildFriendsFragment.java]: can @cr2david address this in a different pr?

**Pull request merged**